src/test/scala/mesosphere/marathon/api/akkahttp/v2/EventsControllerTest.scala

```diff
@@ -0,0 +1,90 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.event.EventStream
+import akka.http.scaladsl.model.RemoteAddress
+import akka.stream.{ ActorMaterializer, ThrottleMode }
+import akka.stream.scaladsl.{ Sink, Source }
+import com.typesafe.config.{ Config, ConfigFactory }
+import mesosphere.UnitTest
+import mesosphere.marathon.api.akkahttp.v2.EventsController.EventStreamSourceGraph
+import mesosphere.marathon.core.event.{ ApiPostEvent, MarathonEvent }
+import mesosphere.marathon.state.{ AppDefinition, PathId }
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{ Millis, Second, Seconds, Span }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class EventsControllerTest extends UnitTest {
+  import EventsControllerTest._
+  implicit val system = ActorSystem("EventsControllerTest", ConfigFactory.load(testConf))
+  implicit val materializer = ActorMaterializer()
+
+  "EventStreamSourceGraph" should {
+    "be a valid source" in {
+      val fakeEventBus: EventStream = new EventStream(system)
+      val sourceUnderTest = Source
+        .fromGraph[MarathonEvent, NotUsed](
+          new EventStreamSourceGraph(fakeEventBus, 5, fakeClientIp)
+        )
+      val fut = sourceUnderTest.take(3).runWith(Sink.seq)
+      Thread.sleep(200) // avoid race with materialization of the stream
```

Comment:
I have rewritten this graph to remove the race. Time-based tests are a recipe for indeterminism.

Check it out in this PR https://github.com/mesosphere/marathon/pull/5548

Reply:
(I'd just go ahead and remove this test :) The tests I wrote test the controller's usage of the stream, and covers this case)

Reply:
Great news, thanks for the heads-up. Will remove the test.